spring-data-querydsl-predicate-injector/src/main/java/com/contentgrid/thunx/spring/data/querydsl/predicate/injector/resolver/OperationPredicates.java

```diff
@@ -0,0 +1,57 @@
+package com.contentgrid.thunx.spring.data.querydsl.predicate.injector.resolver;
+
+import com.querydsl.core.types.Predicate;
+import java.util.List;
+import org.springframework.lang.Nullable;
+
+/**
+ * A set of QueryDSL predicates that apply to different operations that can be executed on an entity
+ */
+public interface OperationPredicates {
+
+    /**
+     * Combines this predicate with a different predicate
+     *
+     * @param predicate The predicate to AND together with
+     * @return The combined predicate
+     */
+    default OperationPredicates and(OperationPredicates predicate) {
+        return new CompositeOperationPredicates(List.of(this, predicate));
+    }
+
+    /**
+     * @return Predicate used for filtering a collection of entities
+     */
+    @Nullable
+    Predicate collectionFilterPredicate();
```

Comment:
Wouldn't returning `Optional<Predicate>`, instead of making everything `@Nullable` be a more appropriate api?